src/monitoring/request_timer.rs

```diff
@@ -0,0 +1,30 @@
+use rocket::fairing::{Fairing, Info, Kind};
+use chrono::Utc;
+use rocket::{Request, Response, Data};
+use rocket::http::Header;
+
+pub struct RequestTimer();
+
+impl Fairing for RequestTimer {
+    fn info(&self) -> Info {
+        Info {
+            name: "RequestTimer",
+            kind: Kind::Request | Kind::Response,
+        }
+    }
+
+    fn on_request(&self, request: &mut Request, _data: &Data) {
+        let instant = Utc::now().timestamp_millis();
+        log::debug!("on_request: {}", &instant);
+        let h = Header::new("requested_timestamp", instant.to_string());
+        request.add_header(h)
+    }
+
+    fn on_response(&self, request: &Request, _response: &mut Response) {
+        let h = request.headers().get("requested_timestamp").next().unwrap().parse::<i64>().unwrap();
+        let instant = Utc::now().timestamp_millis();
+
+        let delta = instant - h;
+        log::debug!("For endpoint {} the request processing duration was {} [ms]", request.uri().path(), delta)
```

Comment:
I would probably use `info` and not debug (but that is only important for the final version)

Reply:
@rmeissner is there a json structure and/or endpoint I need to submit this data for kibana?

Reply:
@Uxio0 needs to help here :D I have no idea ... so this log should already show up but we want to have a format that it is easy to define an Regex (?!?) that can extract the information, so that kibana can handle them and we can build dashboards with the. .... I just have no clue how this works :P

Reply:
Usually we do it using a regex configured in fluentd (the service that's gathering all the logs and providing to Elastic Search), I think that there's the option of providing the JSON to fluentd, but devops are not using it at all as far as I know